From 9a42f183f080fb469e81b36cd02ab3da1b1bedb0 Mon Sep 17 00:00:00 2001
From: "guoyin.chen" <guoyin.chen@freescale.com>
Date: Wed, 21 May 2014 13:32:18 +0800
Subject: [PATCH 501/502] ENGR00314547 Mediaserver crash when no camera been
 connected

Crashed at ~DeviceAdapter() for free an uninitialised mVideoInfo
Fixed by init the mVideoInfo to NULL

Signed-off-by: guoyin.chen <guoyin.chen@freescale.com>
---
 mx6/libcamera2/DeviceAdapter.cpp | 4 ++--
 1 file changed, 2 insertions(+), 2 deletions(-)

diff --git a/mx6/libcamera2/DeviceAdapter.cpp b/mx6/libcamera2/DeviceAdapter.cpp
index 02c1d9e..b78dcf9 100755
--- a/mx6/libcamera2/DeviceAdapter.cpp
+++ b/mx6/libcamera2/DeviceAdapter.cpp
@@ -1,5 +1,5 @@
 /*
- * Copyright (C) 2012-2013 Freescale Semiconductor, Inc.
+ * Copyright (C) 2012-2014 Freescale Semiconductor, Inc.
  *
  * Licensed under the Apache License, Version 2.0 (the "License");
  * you may not use this file except in compliance with the License.
@@ -53,7 +53,7 @@ sp<DeviceAdapter>DeviceAdapter::Create(const CameraInfo& info)
 }
 
 DeviceAdapter::DeviceAdapter()
-    : mCameraHandle(-1), mQueued(0), mCpuNum(0)
+    : mVideoInfo(NULL), mCameraHandle(-1), mQueued(0), mCpuNum(0)
 {}
 
 DeviceAdapter::~DeviceAdapter()
-- 
1.8.0

